tools/cpp_build/build_all.sh

```diff
@@ -4,37 +4,65 @@ set -ex
 
 SCRIPTPATH="$( cd "$(dirname "$0")" ; pwd -P )"
 
-INSTALL_PREFIX="$1"
-
-if [[ -z $INSTALL_PREFIX ]]; then
-  INSTALL_PREFIX=$SCRIPTPATH/build/Install
-fi
-
 cd $SCRIPTPATH
 
-PYTORCHPATH="$SCRIPTPATH/../../"
+INSTALL_PREFIX="${1:-$SCRIPTPATH/build/install}"
+PYTORCHPATH="$SCRIPTPATH/../.."
 
 if [ ! -d "$PYTORCHPATH/torch/csrc/autograd/generated" ]; then
   echo "Generated files are not present.\nRun the generators through `python setup.py build` or copy the generated files over."
   exit 1
 fi
 
+NO_CUDA=ON
 if [ -x "$(command -v nvcc)" ]; then
   NO_CUDA=OFF
-else
-  NO_CUDA=ON
 fi
 
 ATEN_BUILDPATH="$SCRIPTPATH/build/aten-build"
 LIBTORCH_BUILDPATH="$SCRIPTPATH/build/libtorch-build"
 
+# Build with Ninja if available. It has much cleaner output.
+GENERATE="Unix Makefiles"
+MAKE=make
+if [ -x "$(command -v ninja)" ]; then
+  GENERATE=Ninja
+  MAKE=ninja
+fi
+
+# Code is developed a lot more than released, so default to Debug.
+BUILD_TYPE=${BUILD_TYPE:-Debug}
+
+# Try to build with as many threads as we have cores, default to 4 if the
+# command fails.
```

Comment:
The script is run with `set -e` at the top, so I would enclose the subsequent call between a `set +e` / `set -e` pair, to temporarily disable exiting the script in case of failure:
```
set +e
JOBS="$(grep -c '^processor' /proc/cpuinfo)"
set -e
```

Reply:
(otherwise it fails on macOS)